src/gsy_e/models/strategy/update_frequency.py

```diff
@@ -159,11 +165,17 @@ def _populate_profiles(self, area: "Area") -> None:
                 final_rate = find_object_of_same_weekday_and_time(
                     self.final_rate_profile_buffer, time_slot)
 
+            # Hackathon TODO: get rid of self.initial_rate, self.final_rate, self.update_counter
+            # and self.market_slot_added_time_mapping in favor of one object
+            # that keeps track of time_slot: attributes
             self.initial_rate[time_slot] = initial_rate
             self.final_rate[time_slot] = final_rate
 
             self._set_or_update_energy_rate_change_per_update(time_slot)
             write_default_to_dict(self.update_counter, time_slot, 0)
+            if time_slot not in self.market_slot_added_time_mapping:
+                elapsed_seconds = self._elapsed_seconds(area)
+                self.market_slot_added_time_mapping[time_slot] = elapsed_seconds
```

Comment:
IMO this is worth another private method with a docstring that is called in `update_and_populate_price_settings`(because it is not really fitting to the method name here `_populate_profiles`)

Reply:
That definitely makes sense.
I believe that we will soon refactor this module so that both spot and future markets share the same way of calculation,  if we plan on visiting it in the upcoming hackathon, I'll leave it as it, otherwise I will encapsulate this logic.

Reply:
Ok, agreed. Leave it as is for now and we will refactor the whole module later.

Reply:
But can you please do

> IMO this is worth another private method with a docstring that is called in update_and_populate_price_settings(because it is not really fitting to the method name here _populate_profiles)
?
This way we do not oversee this when refactoring. What do you think?